### ISSUE TITLE
L4t graphics demo of EGL image can't display on AGX Orin

### ISSUE THREAD

**Issue by AlexBu** [OE4T/tegra-demo-distro#260]
Hi,

I cloned the repo and build the demo-image-egl image as the readme instructed.
I managed to boot the image on a Jetson AGX Orin devkit 32GB board.
However the l4t graphics demos can't get displayed.

Error log:
```
root@jetson-agx-orin-devkit:/usr/bin/l4t-graphics-demos/egldevice# ./bubble
DRM_CLIENT_CAP_ATOMIC not available.
Could not create the DRM device
<omit program help texts here>

Display_init not yet done[0].
Display_init not yet done[0].
```

I tried all the demos, they are give the same errors.

The eglstreams-kms-example can't display either with error log:
```
root@jetson-agx-orin-devkit:/usr/bin# ./eglstreams-kms-example       
GetDrmFd: drmDeviceFile is /dev/dri/card0
ERROR:  not available.
```

I tried three branches, kirkstone, mickledore and langdale; they all gave the same errors.

**Comment by madisongh** [comment]
From what terminal are you invoking those programs?  The lack of framebuffer console support in L4T R35.x makes the `demo-image-egl` images less useful, so they haven't been very well tested.   There might be a way (using, say, `openvt`) to fire up those applications in a way that EGL can find the display to use, but I haven't really explored that option.  I wouldn't expect that running them from a serial console or ssh terminal to work, though.

**Comment by ichergui** [comment]
Hey @AlexBu 
Any updates about this issue ?

**Comment by AlexBu** [comment]
Well, I turned to Boot 2 Qt image on AGX Orin and didn't follow this case after that. But I did have some progress after all.

The l4t demos under the egldevices platform on the boot 2 qt image did run correctly, though I had no idea why.

I somehow compared the initialization steps between one of the l4t demo and the qt program. And there are two differences.
One is setting a default framebuffer id, which is already been patched on qt5 in meta-tegra layer. The other is setting a valid display attribute.

The display is OK after applying these two changes, under egldevices, of course. GBM platform still has problems, kmscube demo still has texture defects on screen. Something needs to be updated in the display driver, probably.
